synapse/storage/roommember.py

```diff
@@ -533,6 +533,45 @@ def is_host_joined(self, room_id, host):
 
         defer.returnValue(True)
 
+    @cachedInlineCallbacks()
+    def was_host_joined(self, room_id, host):
+        """Check whether the server is or ever was in the room.
+
+        Args:
+            room_id (str)
+            host (str)
+
+        Returns:
+            bool: whether the host is/was in the room or not
```

Comment:
nuh. it returns a deferred.

also: "True if the host is/was in the room. Otherwise False".